extensions/rocksdb-repos/DatabaseContentRepository.cpp

```diff
@@ -198,27 +212,63 @@ bool DatabaseContentRepository::exists(const minifi::ResourceClaim &streamId) {
 }
 
 bool DatabaseContentRepository::removeKey(const std::string& content_path) {
-  if (!is_valid_ || !db_) {
-    logger_->log_error("DB is not valid, could not delete %s", content_path);
-    return false;
-  }
-  auto opendb = db_->open();
-  if (!opendb) {
-    logger_->log_error("Could not open DB, did not delete %s", content_path);
-    return false;
-  }
-  rocksdb::Status status;
-  status = opendb->Delete(rocksdb::WriteOptions(), content_path);
-  if (status.ok()) {
-    logger_->log_debug("Deleting resource %s", content_path);
-    return true;
-  } else if (status.IsNotFound()) {
-    logger_->log_debug("Resource %s was not found", content_path);
-    return true;
-  } else {
-    logger_->log_error("Attempted, but could not delete %s", content_path);
-    return false;
+  if (purge_period_ == std::chrono::seconds(0)) {
+    if (!is_valid_ || !db_)
+      return false;
+    // synchronous deletion
+    auto opendb = db_->open();
+    if (!opendb) {
+      return false;
+    }
+    rocksdb::Status status = opendb->Delete(rocksdb::WriteOptions(), content_path);
+    if (status.ok()) {
+      logger_->log_debug("Deleting resource %s", content_path);
+      return true;
+    } else if (status.IsNotFound()) {
+      logger_->log_debug("Resource %s was not found", content_path);
+      return true;
+    } else {
+      logger_->log_debug("Attempted, but could not delete %s", content_path);
+      return false;
+    }
   }
```

Comment:
Nitpicking, but I would make the inside of this `if` into a separate function.  This way, we have to make sure to have a `return` at the end of every code path; that way, the compiler would check it for us.

Reply:
done